crates/transpiler/src/passes/elide_permutations.rs

```diff
@@ -53,26 +50,29 @@ pub fn run_elide_permutations(
                     mapping.swap(index0, index1);
                 }
                 "permutation" => {
```

Comment:
Just to be clear, since "permutation" isn't a standard gate, that's why this part is kept in Python??

Reply:
Yep exactly, `PermutationGate` is defined in Python and to get the permutation array we need Python to access it. When we have a more general framework for custom operations this will be a good one to move over. Some thought will need to go into how we handle custom attributes like the permutation array here, but for right now and this PR we can safely say that this is the domain of Python.